parser.cpp

```diff
@@ -30,21 +29,81 @@ Result<std::unique_ptr<ast::Str>> Parser::parseStr() {
 }
 
 Result<std::unique_ptr<ast::Out>> Parser::parseOut() {
+  Result<std::unique_ptr<ast::Str>> sres = parseStr();
+  if (sres.hasError()) return sres;
+  std::unique_ptr<ast::Str> &str = sres.get();
+  return Result<std::unique_ptr<ast::Out>>(
+      std::make_unique<ast::Out>(std::move(str)));
```

Comment:
You can just have `Result<std::unique_ptr<ast::Out>>(std::move(str))` since `str` is already a unique_ptr so you don't have to `make_unique` another one.